packages/@react-spectrum/toast/src/ToastContainer.tsx

```diff
@@ -73,9 +74,11 @@ export function ToastContainer(props: SpectrumToastContainerProps): ReactElement
   // We use a ref to do this, since it will have a stable identity
   // over the lifetime of the component.
   let ref = useRef();
-  toastProviders.add(ref);
 
-  // eslint-disable-next-line arrow-body-style
+  useLayoutEffect(() => {
+    toastProviders.add(ref);
+  }, [toastProviders, ref]);
+
```

Comment:
Pretty sure this must be done in render so `useActiveToastContainer` actually returns something. Adding it in the effect also doesn't trigger a re-render.

Reply:
hmm it failed tests when it was in render, and now it passes them. I'll look more closely though, maybe something else is happening or we are missing a test case. Thanks

Reply:
so now this means ToastContainer requires 2 renders on mount. what was breaking in the tests before? I was pretty sure it was safe because `useActiveToastContainer` only changes when there goes from being zero to one toast container, not when any other toast containers are added.

Reply:
It isn't safe because in the cleanup of the useEffect we remove the toastProvider, and we never add it back in for the second firing of effects

Reply:
Ah hmm. I wonder if adding it both in render and in an effect would work. Since it is a set they'd get deduped.

Reply:
So I tried that, and it blew up. Though I'm not really sure why, I can look into that more.
It would result in a different ToastProvider from render to effect though since the Set iterator would essentially do a reverse because the items were added in the reverse order